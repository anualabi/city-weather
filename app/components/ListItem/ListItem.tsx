import {
  Image,
  Pressable,
  StyleProp,
  StyleSheet,
  TextStyle,
  View,
  ViewStyle,
} from "react-native";

import Icon from "@/components/Icon";
import Text from "@/components/Text";

export interface ListItemProps {
  iconColor?: string;
  iconSize?: number;
  image?: string;
  onPress?: () => void;
  showChevron?: boolean;
  style?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  title: string | number;
  title2?: string | number;
}

export default function ListItem({
  iconColor = "black",
  iconSize = 35,
  image,
  onPress,
  showChevron,
  style,
  textStyle,
  title,
  title2,
}: ListItemProps) {
  return (
    <Pressable
      accessibilityRole="button"
      onPress={onPress}
      style={[styles.container, style]}
    >
      {image && (
        <Image
          accessible
          accessibilityLabel="List item image"
          style={styles.image}
          source={{ uri: image }}
          testID="list-item-image"
        />
      )}
      <View style={styles.title}>
        <Text style={[styles.text, textStyle]}>{title}</Text>
        {title2 && <Text style={[styles.text, textStyle]}>{title2}</Text>}
      </View>
      {showChevron && (
        <Icon
          name="chevron-right"
          size={iconSize}
          color={iconColor}
          testID="chevron-icon"
        />
      )}
    </Pressable>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    backgroundColor: "white",
    flexDirection: "row",
    opacity: 0.8,
    padding: 10,
    marginVertical: 8,
  },
  image: {
    height: 50,
    marginRight: 10,
    width: 50,
  },
  text: { fontSize: 18, fontWeight: "bold" },
  title: {
    flexDirection: "row",
    flex: 1,
    justifyContent: "space-between",
  },
});
