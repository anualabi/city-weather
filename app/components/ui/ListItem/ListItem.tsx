import {
  Image,
  Pressable,
  StyleProp,
  StyleSheet,
  TextStyle,
  View,
  ViewStyle,
} from "react-native";

import { colors } from "@/theme";
import Text from "@/components/ui/Text";

export interface ListItemProps {
  onPress?: () => void;
  rightIcon?: React.ReactNode;
  showChevron?: boolean;
  style?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  title: string | number;
  title2?: string | number;
}

export function ListItem({
  onPress,
  rightIcon,
  style,
  textStyle,
  title,
  title2,
}: ListItemProps) {
  return (
    <Pressable
      accessibilityRole="button"
      onPress={onPress}
      style={[styles.container, style]}
    >
      <View style={styles.title}>
        <Text style={[styles.text, textStyle]}>{title}</Text>
        {title2 && <Text style={[styles.text, textStyle]}>{title2}</Text>}
      </View>
      {rightIcon}
    </Pressable>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    backgroundColor: colors.bgWhite,
    flexDirection: "row",
    paddingHorizontal: 20,
    paddingVertical: 20,
  },
  text: { fontSize: 18, fontWeight: "bold" },
  title: {
    flexDirection: "row",
    flex: 1,
    justifyContent: "space-between",
  },
});
