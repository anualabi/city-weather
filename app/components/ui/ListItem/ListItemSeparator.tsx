import { View, StyleSheet } from "react-native";

import { colors } from "@/theme";

export function ListItemSeparator() {
  return <View style={styles.separator} />;
}

const styles = StyleSheet.create({
  separator: {
    backgroundColor: colors.border,
    height: StyleSheet.hairlineWidth,
  },
});
