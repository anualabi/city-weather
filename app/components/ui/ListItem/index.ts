import { ListItem } from "./ListItem";
import { ListItemSeparator } from "./ListItemSeparator";

export { ListItem, ListItemSeparator };
