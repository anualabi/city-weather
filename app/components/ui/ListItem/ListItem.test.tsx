import "@testing-library/jest-native/extend-expect";
import { fireEvent, render } from "@testing-library/react-native";
import renderer from "react-test-renderer";

import { ListItem, ListItemProps } from "./ListItem";

const customRender = (props: ListItemProps) => {
  return render(<ListItem {...props} />);
};

describe("ListItem component", () => {
  test("should render correctly", () => {
    const tree = renderer.create(<ListItem title="Sample Title" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render the title", () => {
    const { getByText } = customRender({ title: "Test Title" });
    expect(getByText("Test Title")).toBeTruthy();
  });

  test("should render secondary title when provided", () => {
    const { getByText } = customRender({
      title: "Test Title",
      title2: "Test Title 2",
    });

    expect(getByText("Test Title 2")).toBeTruthy();
  });

  test("should call onPress when pressed", () => {
    const onPressMock = jest.fn();
    const { getByRole } = customRender({
      title: "Test Title",
      onPress: onPressMock,
    });
    fireEvent.press(getByRole("button"));
    expect(onPressMock).toHaveBeenCalled();
  });
});
