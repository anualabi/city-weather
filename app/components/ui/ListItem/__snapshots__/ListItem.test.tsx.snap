// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component should render correctly 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexDirection": "row",
        "paddingHorizontal": 20,
        "paddingVertical": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#343434",
            "fontSize": 16,
          },
          [
            {
              "fontSize": 18,
              "fontWeight": "bold",
            },
            undefined,
          ],
        ]
      }
    >
      Sample Title
    </Text>
  </View>
</View>
`;
